src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -251,6 +266,13 @@ public MqttClient publish(String topic, Buffer payload, MqttQoS qosLevel, boolea
   @Override
   public MqttClient publish(String topic, Buffer payload, MqttQoS qosLevel, boolean isDup, boolean isRetain, Handler<AsyncResult<Integer>> publishSentHandler) {
 
+    if (countInflightQueue >= options.getMaxInflightQueue()) {
+      publishSentHandler.handle(Future.failedFuture(
+        String.format("Attempt to exceed the limit of %d inflight messages", options.getMaxInflightQueue())
+      ));
```

Comment:
I have just added an `MqttException` class for handling "internal" errors. I have used it for invalid topic name and filter. You should add a new reason code there (i.e. `INFLIGHT_QUEUE_FULL`) and using such exception for the failed future.